rust/template/lib.rs

```diff
@@ -249,13 +318,24 @@ pub unsafe extern "C" fn ddlog_transaction_commit(prog: *const HDDlog) -> raw::c
     res
 }
 
+unsafe fn record_transaction_commit(prog: &sync::Arc<HDDlog>) {
+    if let Some(ref f) = prog.replay_file {
+        if write!(f.lock().unwrap(), "commit;\n").is_err() {
+            prog.eprintln("ddlog_transaction_commit(): failed to record invocation in replay file");
```

Comment:
your error messages are copy-paste; do you want a different message for each failure?

Reply:
They are not exactly the same; each carries the name of the function, which uniquely identifies its location.